File: FreelanceFrontEnd/src/app/types/Freelance-response.ts
import {TechnologyRequest} from "./Technology-request";
import {TechnologyResponse} from "./Technology-response";

export interface FreelanceResponse{
  id:number,
  firstName:string,
  lastName:string,
  userName:string,
  city:string,
  jobTitle:string,
  imageUrl?:string,
  cvUrl?:string,
  technologyResponses?:TechnologyResponse[]
}

File: FreelanceRestApi/src/main/java/com/hamzabekkaoui/freelancerestapi/dto/response/FreelanceResponse.java
package com.hamzabekkaoui.freelancerestapi.dto.response;

import lombok.Builder;

import java.util.List;

@Builder
public record FreelanceResponse(
        String firstName,
        String lastName,
        String userName,
        String city,
        String jobTitle,
        String image,
        List<TechnologyResponse> technologyResponses

) {
}
